python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -328,38 +328,62 @@ def dependencies(self) -> DependencyMapping[NodeInvocation]:
     @public
     @property
     def executor_def(self) -> ExecutorDefinition:
+        """Returns the default :py:class:`ExecutorDefinition` for the job.
+
+        If the user has not specified an executor definition, this will be the :py:func:`multi_or_in_process_executor`, which defaults to multiprocess mode.
+        """
         return self._executor_def or DEFAULT_EXECUTOR_DEF
 
     @public
     @property
     def has_specified_executor(self) -> bool:
+        """Returns True if this job has explicitly specified an executor, and False if the executor was inherited through defaults or the :py:class:`Definitions` object the job was provided to.
```

Comment:
Given the above confusion around "default", would avoid "inherited through defaults" here and just say "inherited from the Definitions object..."